pkg/costattribution/caimpl/managerImpl.go

```diff
@@ -0,0 +1,158 @@
+package caimpl
```

Comment:
to isolate the dependencies, all function needs to be called out side of package should lay in manager.go

Reply:
I personally find this approach strange and unnecessary. Why do we need to isolate that? We don't do that anywhere.

I would move everything into the `costattribution` package.

Reply:
to avoid import cycle, some projects do this, and easy mocks with interface. 👍 since we don't use this in Mimir would remove it to cost attribution as requested.

Reply:
I would suggest building the mocks in a different package, so production code just needs its production package, but test packages would import the mock packages. If you use `mockery` to generate them, you can use `--outpkg` and `--output` flags for that. 

Otherwise you'd need to put your mocks in the production interface package, and import two packages from your prod package.

Reply:
I don't think we've used `mockery` so far, but I'm not telling you to use _mockery_, the tool you use for the mock is orthogonal to the mock placement.